src/handlers/platformProviderResponsesHandlers.ts

```diff
@@ -0,0 +1,124 @@
+import { ajv } from '../ajv';
+import { db } from '../database';
+import { isTinyPgErrorWithQueryContext } from '../types';
+import {
+  DatabaseError,
+  InputValidationError,
+} from '../errors';
+import type {
+  Request,
+  Response,
+  NextFunction,
+} from 'express';
+import type { JSONSchemaType } from 'ajv';
+import type { PlatformProviderResponse } from '../types';
+
+interface GetPlatformProviderResponseByExternalIdParams {
+  externalId: string;
+}
+const getPlatformProviderResponsesByExternalIdParamsSchema
+: JSONSchemaType<GetPlatformProviderResponseByExternalIdParams> = {
+  type: 'object',
+  properties: {
+    externalId: {
+      type: 'string',
+      pattern: '^\\d{9}$',
```

Comment:
This is the "we're storing EINs without dashes here" choice you called out verbally.

Just to talk through what I was suggesting: right now, in prod, every EIN we're storing in the `organization_tax_id` field has a dash (including the invalid ones!). Candid supports looking up by EINs that include a dash. My plan for the Candid data script in https://github.com/PhilanthropyDataCommons/data-scripts/pull/24 is to get the EIN from the proposal, look it up in Candid, and then upload the Candid response - without much parsing - to the PDC via the new write API.

Later, the data-viewer will make a request in the same fashion - get the EIN from the `organization_tax_id` field, and make a request to the new read API.

Requiring dashless EINs adds extra processing to every client, and I'm not clear on what the benefit is - can you talk more about the upside?

Reply:
To be clear: it doesn't *quite*matter what candid's API supports since what we're defining here is an API that will only support that if we write the logic to convert the input provided.  To that end, it is much easier to just strip dashes, which means dash stripping is going to happen in order to normalize the storage / query (we don't want a situation where the candid data for an org is against a different external ID than the other data platform providers for that same org).

Which is to say: the processing has to happen regardless for every query / we are going to want to store without dashes.  It felt a little odd to me for someone to POST `{ externalId: '11-1111-111' }` and get back a modified value (`{ externalId: '111111111' }`) as opposed to requiring the client to adhere to a clearly documented API spec.

Does that make sense?  Happy to talk by voice about it if that's easier.

Reply:
... actually, maybe the solution here is to not use external ID in the first place, and instead store this in relation to an applicant via an applicant_id foreign key.

Reply:
After discussion, I think we arrived at the decision that we do _not_ want to link this to applicants; this is a separate, standalone, tech-debt table that we are writing with the intention to delete it when we have the opportunity to build this feature out more fully and with less deadline pressure. Pilot, not MVP!

We also don't know what that fuller feature is going to look like: maybe there'll be a way to connect your data platform provider account? Maybe data platform providers will push data rather than have us pull data? Probably we'll map to our base fields?

So, for now, we're doing a solely application-side foreign key (that is, not in the database, just in client code), and we needn't normalize at all - whatever external ID the [data-scripts] client posts is what we save, and whatever external ID the [data-viewer] client looks up is what we query by, and if those differ than that's very sad.

Reply:
(Thank you for taking the time to summarize -- +1 to all you wrote!)